.github/workflows/ios.yml

```diff
@@ -64,6 +64,13 @@ jobs:
                   APPLE_DEMO_EMAIL: ${{ secrets.APPLE_DEMO_EMAIL }}
                   APPLE_DEMO_PASSWORD: ${{ secrets.APPLE_DEMO_PASSWORD }}
 
+            - name: Run Fastlane for App Store release
+              if: ${{ env.SHOULD_DEPLOY_PRODUCTION }}
+              run: bundle exec fastlane ios production
+              env:
+                # TODO: Make sure this is the iOS version e.g. 1.0.2.2, not JS version 1.0.2-2
```

Comment:
@roryabraham - Not sure if you know if `NEW_VERSION` will be JS or iOS style, but if it's in the JS style we can convert to iOS style in ruby.

Reply:
So eventually this workflow will need to also be triggered by the creation of a GH release, and not just pushing tags. If it's triggered by a GH release, then `SHOULD_DEPLOY_PRODUCTION` will be true. Otherwise (triggered by pushing tags), it will be false.

If you need to provide the version as input for `deliver` stuff, we should be able to easily get it from the github context. Since it will have been triggered by the creation of a release, the version should be available at 
`${{ github.event.tag_name }}`. It will be the JS style, but also will only ever have three parts `1.0.3`.

Also just a note for the future, we'll have to make sure _not_ to trigger this as part of a `CP Production`, because we know it won't work.

Reply:
Why not have some string interpolation to replace `-` with `.` ?

Reply:
@roryabraham 

> If it's triggered by a GH release, then `SHOULD_DEPLOY_PRODUCTION` will be true. Otherwise (triggered by pushing tags), it will be false.

Did you want me to add that logic now?

> It will be the JS style, but also will only ever have three parts `1.0.3`.

I believe we need the build number in order to tell Apple which version should go live.

@Dal-Papa 

> Why not have some string interpolation to replace `-` with `.` ?

Agreed, we will probably have to add in some of this logic, but we are trying to decide if it's 100% necessary.

Reply:
> Did you want me to add that logic now?

No, I think for now just creating the lane is fine. We can add that logic as part of https://github.com/Expensify/Expensify/issues/155232

What do you think?

Reply:
Sounds good, but are you sure we will only have the short version? That is a requirement for this to work at the moment.

Reply:
> I believe we need the build number (full version 1.0.2.3) in order to tell Apple which version should go live.

Hmmm this might be problematic, because when we deploy to production we'll bump the `PATCH` version `1.0.2-3` -> `1.0.3`, and won't have a `BUILD` version really. So the `CFBundleVersion` would also just be `1.0.3`. If we need to, we can default the `BUILD` version to `0`.

Reply:
> we can default the BUILD version to 0.

We could do that by changing [this line](https://github.com/Expensify/Expensify.cash/blob/0e240f726ed9e95f638e2ea52ac25438354cd8aa/.github/libs/nativeVersionUpdater.js#L48) to:

```javascript
const cfVersion = version.includes('-') ? version.replace('-', '.') : `${version}.0`;
```

Reply:
Bumping from `1.0.2-3` -> `1.0.3-0` (iOS: `1.0.3.0`) I think would be great 👍

Reply:
Should we update this todo comment w/ the link to https://github.com/Expensify/Expensify/issues/156562?

Reply:
Added it as a todo to the linked issue 👍